geode-docs/developing/transactions/design_considerations.html.md.erb

```diff
@@ -0,0 +1,152 @@
+or configure the system to return copies instead of references.
+There is a performance penalty to having the system return copies.
+Both ways are detailed in [Safe Entry Modification](../../basic_config/data_entries_custom_classes/managing_data_entries.html#managing_data_entries__section_A0E0F889AC344EFA8DF304FD64418809).
+
+## <a id="first-op-with-mixed-types" class="no-quick-link"></a>First Operation with Mixed Region Types
+
+When more than one region participates in a transaction,
+and there is at least one partitioned and at least one
+replicated region,
+the code must do its first operation on the partitioned
+region to avoid a `TransactionDataNotColocatedException`.
+Write the transaction to do its first operation on a partitioned
+region, even if the operation will be spurious.
+
+## <a id="transactions-persistence" class="no-quick-link"></a> Allowing Transactions to Work on Persistent Regions
+
+<%=vars.product_name%>'s implementation of the durability promise prohibits
+regions with persistence from participating in transactions.
+The invocation of a persistent region operation within a transaction
+throws an `UnsupportedOperationException` with an associated message of
+
+``` pre
+Operations on persist-backup regions are not allowed because this thread
+has an active transaction
+```
+
+An application that wishes to allow operations on a persistent region during
+a transaction can set this system property:
+
+`-Dgemfire.ALLOW_PERSISTENT_TRANSACTIONS=true`
+
+Setting this system property eliminates the exception.
+It does not provide durability.
+
+## <a id="transactions-indexes" class="no-quick-link"></a>Mixing Transactions with Indexes
+
+The contents and updates to an index do not intersect with any
+changes made within a transaction.
+Any operation on an index done within a transaction
+will not reflect any state from within that transaction.
+Therefore, do not mix transactions with operation on an index.
+
+## <a id="transactions-queries" class="no-quick-link"></a>Mixing Transactions with Queries
+
+Queries and query results reflect region state, and not any state or
+changes that occur within a transaction.
+Therefore, do not mix transactions with queries.
+
+## <a id="transactions-eviction" class="no-quick-link"></a>Mixing Transactions with Eviction
+
+LRU eviction and transactions work well together.
+Any eviction operation on a region entry that is operated on
+from within a transaction is deferred until the transaction is committed.
+Further, because any entry touched by the transaction
+has had its LRU clock reset,
+eviction is not likely to choose those entries as victims
+immediately after the commit.
+
+## <a id="transactions-expiration" class="no-quick-link"></a>Mixing Transactions with Expiration
+
+A transaction disables expiration on any region entries affected
+by the transaction.
+
+## <a id="transactions-multithreaded" class="no-quick-link"></a>Mixing Transactions with Multi-threaded Applications
```

Comment:
This info probably belongs in the Isolation section of  the ACID page, talking about how to handle potential dirty reads. I wouldn't call it "Mixing with multi-threaded applications" - this is really about how to handle read conflicts.